### DISCUSSION TITLE
Issues about simulation with ZIP model

### DISCUSSION THREAD

**Discussion by Albizzian** [CURENT/andes#542] [Q&A] [answered]
I add a ZIP model to the system, however the transient simulation results don't change before and after adding.  Could you please help me to check what's wrong with it?
[ieee39_REGCV1_35%.xlsx](https://github.com/CURENT/andes/files/15417225/ieee39_REGCV1_35.xlsx)
![image](https://github.com/CURENT/andes/assets/78994912/5704ca65-e908-4be1-b4fd-be4ecff37596)
![image](https://github.com/CURENT/andes/assets/78994912/b9ee3871-b01b-4a9a-a58f-bf1a6d2895b6)

**Comment by cuihantao** [comment, marked as answer]
Can you post the script in "```" blocks for easy copy?

The impact of the disturbance on machine speed is quite small. Could it be that the ZIP model was successfully replaced, but the difference is tiny?


You can check power injection of the ZIP device after TDS initialization but before TDS runs. Check `ss.ZIP.a.e` and `ss.ZIP.v.e`, the injective active power and reactive power, to ensure that they match the original PQ model. If it checks out, run the simulation and check the two fields. You should see slightly different values after the disturbance. If that's the case, the ZIP successfully replaces the PQ, and not seeing the difference is expected.

You can use a more severe fault with a larger impact on voltage to observe the effect fo ZIP. The voltage dip should be more mild with ZIP.

**Reply by Albizzian** [reply]
Thanks for your advice!  Here is the script:
```
ss=andes.load('ieee39_REGCV1_35%_ZIP.xlsx',setup=False)
ss.add("Alter",dict(model='PQ',dev='PQ_16',src='Ppf', t=1.0,method='*',amount=1.5))
ss.add("ZIP",dict(pq='PQ_14',kpp=40,kpi=30,kpz=30,kqp=40,kqi=30,kqz=30))
ss.PQ.config.p2p = 1.0
ss.PQ.config.p2i = 0
ss.PQ.config.p2z = 0
ss.PQ.config.q2q = 1.0
ss.PQ.config.q2i = 0
ss.PQ.config.q2z = 0 
ss.PQ.config.pq2z = 0
ss.TDS.config.tstep=1/200
ss.setup()
ss.PFlow.run()
ss.TDS.config.tf=10 
ss.TDS.config.criteria=0 
ss.TDS.run()
```
And I check  `ss.ZIP.a.e` and `ss.ZIP.v.e` before TDS runs, but they are all empty. It seems that the ZIP doesn't successfully replace the PQ?

**Reply by cuihantao** [reply]
Before TDS runs but after `TDS.init()`. Check again?

**Reply by Albizzian** [reply]
I added `ss.TDS.init()` and checked again. Small changes of active power and reactive power of ZIP can be observed.  It indicates that  ZIP model successfully replaces PQ model. I think you're right that the impact of the disturbance on machine speed is too small to observe obvious difference.
Thanks for your kindly and detailed reply again!
